### ISSUE TITLE
Techdocs "recommended approach" implementation ideas

### ISSUE THREAD

**Issue by shanthisagar-dhanya-kumar-db** [backstage/backstage#10543]
Currently in our organisation, the technical docs are all scattered across, as it resides in the entity owners' git repos which we don't have control over. But, we in the backstage team maintain some of the backstage yaml file referencing those techdocs.

If we were to use CI/CD to build the techdocs, we will need to download (clone) all the docs content of docs to CI/CD server and then further execute techdocs-cli. Even before that we would need to know all the locations of all the docs in our organisation which is still possible via https://XXXXX/api/catalog/entities?filter=metadata.annotations.backstage.io%2Ftechdocs-ref. 
Some of the entities' techdocs' annotations are relative reference dir:. and some are absolute with url:.

So, basically to execute the techdocs-cli generate, we would need to have all the contents stored on the server where we are trying to generate docs. 
Is there a way we just pass the url reference of the location where the docs are stored to the techdocs-cli and it does the rest of downloading the content and generating the docs OR is it expected the we build all this capability in the CI/CD pipeline?

**Comment by garyniemen** [comment]
@shanthisagar-dhanya-kumar-db At Spotify, each team does its own CI/CD so each team builds its own doc site and it all ends up in Backstage. Are you saying  that in your company, everything is built centrally? Can you say more.

**Comment by shanthisagar-dhanya-kumar-db** [comment]
We don't have that culture built yet but I think we need to be where you guys are.
At the moment, we build everything for the while organization which could potentially change.
